Added BombCrypto adapter on the Marketplace

ankitzm opened the pull request:
1. Added abi.json for Contract address [0xe29F0B490F0d89CA7ACAc1C7BeD2E07eCAD65201](https://www.bscscan.com/address/0xe29f0b490f0d89ca7acac1c7bed2e07ecad65201)
2. Added index.js with all the required information.

Bomb Crypto Marketplace - https://market.bombcrypto.io/ (On BNB)
Contract Address - 0xe29F0B490F0d89CA7ACAc1C7BeD2E07eCAD65201

Other Proxy Contracts :arrow_down: 
- https://www.bscscan.com/address/0x376A10E7f125A4E0a567cc08043c695Cd8EDd704
- https://www.bscscan.com/address/0x049896f350C802CD5C91134E5f35Ec55FA8f0108

ankitzm commented:
Hey @liutaurasjanuskas, it would be great if you can review my PR and/or provide me with the API key to test my adapter. 
I have already filled the Google Form for getting the API key but haven't received any mail. :mailbox_with_no_mail:

liutaurasjanuskas commented on adapters/bombcrypto-bnb/index.js:
```diff
@@ -0,0 +1,148 @@
+        this.token = "bnb";
+        this.protocol = "binance";
+        this.block = 8867059;
+        this.contract = "0xe29F0B490F0d89CA7ACAc1C7BeD2E07eCAD65201";
```
All EVM chains (Ethereum, Polygon, BSC, ...) contracts must be lowercased

ankitzm replied:
Thanks for the review Liutauras. I have converted the Contract Address to lowercase.

ankitzm commented:
Thanks for the API key over mail @liutaurasjanuskas.
I have tested the adapter and found some errors in my code. Fortunately I was able to fix them and pushed those changes in the repo. It would be great if you can review the code now.

**Changes**
1. Got ABI with all the functions from the Proxy Contract and replaced the old ABI file.
2. Original Contract Address is 0xe29F0B490F0d89CA7ACAc1C7BeD2E07eCAD65201 and you can check about the proxy contract address with this [link](https://bscscan.com/address/0xe29F0B490F0d89CA7ACAc1C7BeD2E07eCAD65201#readProxyContract).
3. Fixed all the function as per new ABI file.
4. Run Test and got the "Test Passed" log.

**Test**

Running the test from the very first block was taking a very long time, so I tested this out from a bit recent block. The block number for contract creation can be found [here](https://bscscan.com/txs?a=0xe29F0B490F0d89CA7ACAc1C7BeD2E07eCAD65201&f=5), it's `8867059`.

I have run the test from `19077059` this block number and it successfully passed !!

![Screenshot from 2022-07-03 13-52-20](https://user-images.githubusercontent.com/66105983/177031920-d458b2e5-bace-47e3-960b-0ac63be446b0.png)

liutaurasjanuskas commented on adapters/bombcrypto-bnb/index.js:
```diff
@@ -0,0 +1,145 @@
+        this.name = "bombcrypto-bnb";
+        this.symbol = "BNB";
+        this.token = "bnb";
+        this.protocol = "binance";
```
We don't support this value as a `protocol`. Change it to `bsc`, please.

ankitzm replied:
Yeah missed that, fixed it now

liutaurasjanuskas commented on adapters/bombcrypto-bnb/index.js:
```diff
@@ -0,0 +1,145 @@
+        const po = await this.getPrice(block.timestamp);
+
+        let value = 0;
+        if (event.event === "ItemListed") {
```
What is the point of having this condition here? In both cases it's setting `value` as `event.returnValues.price`

ankitzm replied:
Removed the else case

liutaurasjanuskas commented on adapters/bombcrypto-bnb/index.js:
```diff
@@ -0,0 +1,145 @@
+        this.protocol = "binance";
+        this.block = 8867059;
+        this.contract = "0xe29f0b490f0d89ca7acac1c7bed2e07ecad65201"; // abi used for 0x5713Ae21F4Bb696A877c90CCcAE310eFF4c4652A(Proxy Contract abi) - https://bscscan.com/address/0xe29F0B490F0d89CA7ACAc1C7BeD2E07eCAD65201#readProxyContract
+        this.events = ["OfferMatched", "ItemListed", "OwnershipTransferred"];
```
Are you sure that `ItemListed` and `OwnershipTransferred` events are necessary here? It's pretty self-explanatory, that it's not making any sale here.

ankitzm replied:
ItemListed is used to get price in "_getPrice" function
OwnershipTransferred event is used in "getBuyer" function, cause it will be transferred if someone buys the NFT.

ankitzm commented:
Het @liutaurasjanuskas, it would be great if you can review the code and let me know if it need some more changes.

liutaurasjanuskas commented on src/adapters/bombcrypto-bnb/index.js:
```diff
@@ -0,0 +1,138 @@
+        this.protocol = "binance-smart-chain";
+        this.block = 8867059;
+        this.contract = "0xe29f0b490f0d89ca7acac1c7bed2e07ecad65201"; // abi used for 0x5713Ae21F4Bb696A877c90CCcAE310eFF4c4652A(Proxy Contract abi) - https://bscscan.com/address/0xe29F0B490F0d89CA7ACAc1C7BeD2E07eCAD65201#readProxyContract
+        this.events = ["OfferMatched", "ItemListed", "OwnershipTransferred"];
```
I am pretty sure that `ItemListed` and `OwnershipTransferred` events are useless here.

ankitzm replied:
Yeah, the code is working without them. Fixed this, I got a bit confused previously.

Mintiss commented:
Marketplace has been integrated
https://github.com/dappradar/nft-sales-adapters/commit/e12984d9833f1311260121a3847ce4785d80b44a
